ape_hardhat/__init__.py

```diff
@@ -0,0 +1,237 @@
+                    "Hardhat process is running, but could not connect to RPC server. "
+                    "Run `npx hardhat node` or adjust retry strategy configs to troubleshoot."
+                )
+            raise HardhatSubprocessError(
+                "Hardhat command exited prematurely. Run `npx hardhat node` to troubleshoot or "
+                "adjust retry strategy configs to troubleshoot."
+            )
+        elif process.poll() is not None:
+            raise HardhatSubprocessError(
+                "Hardhat process exited prematurely, but connection succeeded. "
+                "Is something else listening on the port?"
+            )
+        self.process = process
+
+    def connect(self):
+        """Start the hardhat process and verify it's up and accepting connections."""
+
+        if self.config.port:
+            # if a port is configured, only make one start up attempt
+            self._start_process()
+        else:
+            for _ in range(self.config.process_attempts):
+                try:
+                    self._start_process()
+                    break
+                except HardhatSubprocessError as exc:
+                    print("Retrying hardhat subprocess startup:", exc)
+
+        # subprocess should be running and receiving network requests at this point
+        if not (self.process and self.process.poll() is None and self.port):
+            raise HardhatSubprocessError(
+                "Could not start hardhat subprocess on a random port. "
+                "See logs or run `npx hardhat node` or adjust retry strategy configs to "
+                "troubleshoot."
+            )
+
+    @property
+    def uri(self) -> str:
+        uri = getattr(self.config, self.network.ecosystem.name)[self.network.name]["uri"]
+        if uri != DEFAULT_SETTINGS["uri"]:
+            # the user configured their own URI in the project configs, let's use that
+            return uri
+        elif self.port:
+            # the user did not override the default URI, and we have a port
+            # number, so let's build the URI using that port number
+            return f"http://localhost:{self.port}"
+        else:
+            raise RuntimeError("Can't build URI before `connect` is called.")
+
+    @property
+    def _web3(self):
+        """
+        This property overrides the ``EthereumProvider._web3`` class variable to return our
+        instance variable.
+        """
+        return self._hardhat_web3
+
+    @_web3.setter
+    def _web3(self, value):
+        """
+        Redirect the base class's assignments of self._web3 class variable to our instance variable.
+        """
+        self._hardhat_web3 = value
+
+    def disconnect(self):
+        super().disconnect()
+        if self.process:
+            # process object exists, try killing it
+            self.process.kill()
+        self.process = None
+
+    def _make_request(self, rpc: str, args: list) -> Any:
```

Comment:
I wonder if it'd be worth making this method public so that if `hardhat` ever adds more network methods, users can feel comfortable calling them by their string names in their script (until we support the method natively).

Reply:
There already seems to be a handful of methods listed here https://hardhat.org/hardhat-network/reference/#evm-setintervalmining
that we don't have in here